app/models/project_step.rb

```diff
@@ -196,6 +206,67 @@ def week_of_scheduled_date
     end
   end
 
+  # todo: confirm if we want to hardcode a limit for end_date bound repeats. and if so, what should that be?
+  MAX_OCCURRENCES = 100
+
+  def duplicate_series(frequency, time_unit, month_repeat_on, num_of_occurrences, end_date)
+    if time_unit == 'days' || time_unit == 'weeks'
```

Comment:
Yeah, this was actually irrelevant code which should have been removed after some refactoring.